### ISSUE TITLE
Autoload directories for package managers is not load when using `--commands`

### ISSUE THREAD

**Issue by doubleailes** [nushell/nushell#13635]
### Describe the bug

The feature Autoload directories don’t load the files for the process run in the `—commands`

### How to reproduce

1. Run `python3.12 -m venv —copies /new`
2. Create a directory /tmp/autoload
3. Create file `toto.nu`
4. Set a new path for bin like `$env.PATH = (&env.PATH | prepend "/new/bin/")` in `toto.nu`
5.  Create a custom `/tmp/env.nu`
6. Set the var $env.NU_VENDOR_AUTOLOAD_DIR= /tmp/autoload in `env.nu`
7. Run `nu —env-config /tmp/env.nu -c python

### Expected behavior

The version of python should the be first one source in the PATH variable python 3.12 in my case and not the version run by the system

### Configuration

| key                | value                                                         |
| ------------------ | ------------------------------------------------------------- |
| version            | 0.96.1                                                        |
| major              | 0                                                             |
| minor              | 96                                                            |
| patch              | 1                                                             |
| branch             |                                                               |
| commit_hash        | f7d6c28a001f94f224e459e21df0da2ab5bdc923                      |
| build_os           | linux-x86_64                                                  |
| build_target       | x86_64-unknown-linux-musl                                     |
| rust_version       | rustc 1.77.2 (25ef9e3d8 2024-04-09)                           |
| rust_channel       | 1.77.2-x86_64-unknown-linux-gnu                               |
| cargo_version      | cargo 1.77.2 (e52e36006 2024-03-26)                           |
| build_time         | 2024-07-29 23:38:40 +00:00                                    |
| build_rust_channel | release                                                       |
| allocator          | mimalloc                                                      |
| features           | default, sqlite, static-link-openssl, system-clipboard, trash |
| installed_plugins  |                                                               |

**Comment by doubleailes** [comment]
Thanks @sholderbach, can you elaborate in which use case you wanna run `nu -c` and be sure to not autoload.

I’m working on the integration of Nushell into a package manager and it useful to run `<exe> env <package> — <command>` which require to source files to run the command into the expected requirements. If you have a work around i’m interest also.

**Comment by sholderbach** [comment]
Not my primary area of expertise, paging @fdncred who spearheaded the autoload stuff.

**Comment by fdncred** [comment]
[Config file loading rules](https://gist.github.com/fdncred/b87b784f04984dc31a150baed9ad2447/) - I think it still works this way. Notice that autoloading is not included here simply because it's a new feature. We could add the functionality, but we'd need to decide how it needs to work related to these rules.

**Comment by doubleailes** [comment]
Thanks @fdncred 

The config file rules is really helpful.

That’s what i hoped about the autoload feature. I stay tuned.

**Comment by doubleailes** [comment]
Hello any news about this?

**Comment by doubleailes** [comment]
Just a little up on this issue.

**Comment by fdncred** [comment]
We've documented how it should work here https://www.nushell.sh/book/configuration.html#detailed-configuration-startup-process

I think that's the link you're looking for.

**Comment by doubleailes** [comment]
Thanks for pointing the documentation but the behavior is still the same.
Running `-c` argument does not use the auto load dir feature. And i don’t any mention of this in the documentation.
How can i help?

**Comment by fdncred** [comment]
Correct. Documentation wasn't meant to fix the problem but explain to you how it's supposed to work.

It mentions autoload

![Image](https://github.com/user-attachments/assets/54c1db57-7384-42a1-8981-f534d0b726ac)

as well as explains that `nu -c` doesn't load these files

![Image](https://github.com/user-attachments/assets/682799c4-04e9-40df-83b9-b6063ef77f1c)

I'd try adding `-l` and see if that fixes your problem.

**Comment by doubleailes** [comment]
Thanks for pointing the part mentioning the autoload and command.

I try `-l` but it is the same.

Reading the documentation it is not clear for me, if `--env-config <path>` is a supported feature with `-c`

**Comment by doubleailes** [comment]
I have test if the python invoked by the command `-c` get access to var. And yes.

```python
>>> import os
>>> print(os.getenv("NU_VENDOR_AUTOLOAD_DIR", "Empty"))
/tmp/autoload
```

**Comment by fdncred** [comment]
`NU_VENDOR_AUTOLOAD_DIR` was meant to be used at compile time to set autoload dirs
